ceph/ceph_admin/dashboard.py

```diff
@@ -112,3 +133,121 @@ def validate_enable_dashboard(cls, username, password):
+        for daemon in daemon_obj:
+            host = daemon["hostname"]
+
+        # To get ip address of that host
+        for node in cls.cluster.get_nodes():
+            if node.hostname == host:
+                node_ip = node.ip_address
+                break
+
+        ALERTMANAGER_ENABLE_COMMANDS = [
+            f"ceph dashboard set-alertmanager-api-host 'http://{node_ip}:{port}'",
+            f"ceph dashboard set-alertmanager-api-ssl-verify {ssl}",
+        ]
+
+        execute_commands(cls, ALERTMANAGER_ENABLE_COMMANDS)
+
+        url, _ = cls.shell(args=["ceph ", "dashboard ", "get-alertmanager-api-host "])
+        validate_url(url)
+    else:
+        LOG.error("Daemon alertmanager does not exists")
+
+
+def enable_prometheus(cls, config):
+    """Method to enable prometheus.
+    Args:
+        cls (CephAdmin object) : cephadm instance object.
+        config (Dict): Key/value pairs passed from the test suite.
+    """
+    port = config.get("prometheus-port")
+    ssl = config.get("prmoetheus-ssl-set")
+
+    # check if prometheus daemon exist
+    out, _ = cls.shell(
+        args=["ceph", "orch", "ps", "--daemon_type", "prometheus", "-f", "json"]
+    )
+    daemon_obj = loads(out)
+
+    if daemon_obj:
+        # To get hostname in which daemon got deployed
+        for daemon in daemon_obj:
+            host = daemon["hostname"]
+
+        # To get ip address of that host
+        for node in cls.cluster.get_nodes():
+            if node.hostname == host:
+                node_ip = node.ip_address
+                break
+
+        PROMETHEUS_ENABLE_COMMANDS = [
+            f"ceph dashboard set-prometheus-api-host 'http://{node_ip}:{port}'",
+            f"ceph dashboard set-prometheus-api-ssl-verify {ssl}",
+        ]
+
+        execute_commands(cls, PROMETHEUS_ENABLE_COMMANDS)
+
+        url, _ = cls.shell(args=["ceph", "dashboard", "get-prometheus-api-host"])
+        validate_url(url)
+    else:
+        LOG.error("Daemon prometheus does not exists")
+
+
+def enable_grafana(cls, config):
+    """Method to enable grafana.
+    Args:
+        cls (CephAdmin object) : cephadm instance object.
+        config (Dict): Key/value pairs passed from the test suite.
+    """
+    port = config.get("grafana-port")
+    ssl = config.get("grafana-ssl-set")
+
+    # check if grafana daemon exist
+    out, _ = cls.shell(
+        args=["ceph", "orch", "ps", "--daemon_type", "grafana", "-f", "json"]
+    )
+    daemon_obj = loads(out)
+
+    if daemon_obj:
+        # To get hostname in which daemon got deployed
+        for daemon in daemon_obj:
+            host = daemon["hostname"]
+
+        # To get ip address of that host
+        for node in cls.cluster.get_nodes():
+            if node.hostname == host:
+                node_ip = node.ip_address
+                break
+
+        GRAFANA_ENABLE_COMMANDS = [
+            f"ceph dashboard set-grafana-api-url 'http://{node_ip}:{port}'",
```

Comment:
How do you plan to handle `https`

Reply:
In dashboard as of now only grafana URL has **HTTPS** and other promotheus and alertmanager are still in **HTTP** based
there is bug on it all other monitoring stack will also goes to https may be in 6.0 release 
As of now I will keep what is supported from dashboard.
Thanks for the printout!